File: app/src/main/java/com/loopeer/codereader/ui/decoration/AppbarBehavior.java
package com.loopeer.codereader.ui.decoration;

import android.content.Context;
import android.support.design.widget.AppBarLayout;
import android.support.design.widget.CoordinatorLayout;
import android.util.AttributeSet;
import android.view.View;

public class AppbarBehavior extends AppBarLayout.Behavior {

    public AppbarBehavior() {
    }

    public AppbarBehavior(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public boolean onStartNestedScroll(CoordinatorLayout parent, AppBarLayout child, View directTargetChild, View target, int nestedScrollAxes) {
        boolean result = super.onStartNestedScroll(parent, child, directTargetChild, target, nestedScrollAxes);
        if (result && !target.canScrollVertically(1))
            result = false;
        return result;
    }
}

File: code-reader-kt/src/main/java/com/loopeer/codereaderkt/ui/decoration/AppbarBehavior.kt
package com.loopeer.codereaderkt.ui.decoration

import android.content.Context
import android.support.design.widget.AppBarLayout
import android.support.design.widget.CoordinatorLayout
import android.util.AttributeSet
import android.view.View


class AppbarBehavior : AppBarLayout.Behavior {

    constructor() {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    override fun onStartNestedScroll(parent: CoordinatorLayout?, child: AppBarLayout?, directTargetChild: View?, target: View?, nestedScrollAxes: Int): Boolean {
        var result = super.onStartNestedScroll(parent, child, directTargetChild, target, nestedScrollAxes)
        if (result && !target!!.canScrollVertically(1) && child!!.y >= 0)
            result = false
        return result
    }
}